src/averell/core.py

```diff
@@ -32,36 +31,68 @@ def get_corpora(corpus_indices=None, granularity=None,
-                granularities_list = CORPORA_SOURCES[index]["properties"][
-                    "granularity"]
-                if granularity in granularities_list:
-                    for poem in features:
-                        filtered_features = filter_features(poem, index,
-                                                            granularity)
-                        granularity_path = gen_path / granularity / poem[
-                            "author"].replace(" ", "")
-                        if filtered_features:
-                            if not granularity_path.exists():
-                                os.makedirs(granularity_path)
-                            write_json(filtered_features, str(
-                                granularity_path / poem[
-                                    "poem_title"].title().replace(" ", "")))
-                            corpora_features.append(filtered_features)
-                else:
-                    corpus_name = CORPORA_SOURCES[index]["name"]
-                    logging.error(
-                        f"'{granularity}' granularity not found on "
-                        f"'{corpus_name}' properties")
             else:
                 corpora_features.append(features)
     except IndexError:
         logging.error("Index number not in corpora list")
     finally:
         return corpora_features
+
+
+def export_corpora(corpus_ids, granularity, corpora_folder):
+    """
+    Generates a single JSON file with the chosen granularity for all of the
+    selected corpora
+    :param corpus_ids: IDs of the corpora that will be exported
+    :param granularity: Level of parsing granularity
+    :param corpora_folder: Local folder where the corpora is located
+    :return: Python dict with the chosen granularity for all of the selected
+        corpora
+    """
+    corpora_features = []
+    if Path(corpora_folder).exists():
+        if not corpus_ids:
+            logging.error("No CORPUS ID selected")
+        else:
+            if granularity is not None:
+                for index in corpus_ids:
+                    corpus_id = index - 1
+                    try:
+                        corpus_folder = CORPORA_SOURCES[corpus_id][
+                            "properties"]["folder_name"]
+                    except IndexError:
+                        logging.error("ID not in corpora list")
+                    else:
+                        if not (Path(corpora_folder) / corpus_folder).exists():
+                            logging.error(
+                                f'{CORPORA_SOURCES[corpus_id]["name"]}'
+                                f' not downloaded')
+                            continue
+                        granularities_list = CORPORA_SOURCES[corpus_id][
+                            "properties"]["granularity"]
+                        if granularity not in granularities_list:
+                            corpus_name = CORPORA_SOURCES[corpus_id]["name"]
```

Comment:
Would this chunk become cleaner if we assigned `corpus = CORPORA_SOURCES[corpus_id]` at the beginning? Just wondering.

Reply:
Yes, like `corpus_folder = CORPORA_SOURCES[corpus_id]["properties"]["folder_name"]`, and assign all the variables at the beginning?